roles/idrac_gather_facts/tests/asserts/controller_assert.yml

```diff
@@ -10,30 +11,67 @@
     body_format: "{{ idrac_gather_facts_uri_body_format }}"
     status_code: "{{ idrac_gather_facts_uri_status_code }}"
     return_content: "{{ idrac_gather_facts_uri_return_content }}"
-  register: controller_result
+  register: storage_entity
   no_log: true
 
-- name: Set controller facts
+- name: Get storage entities
   ansible.builtin.set_fact:
-    api_response: "{{ controller_result.json | ansible.utils.remove_keys(target=['@odata.context', '@odata.id', '@odata.type']) }}"
+    collected_storage_entity: "{{ (collected_storage_entity | default([])) + [item['@odata.id']] }}"
+  loop: "{{ storage_entity.json.Members }}"
 
-- name: Set the keys diff
+- name: Get Controllers URL information.
+  ansible.builtin.uri:
+    url: "https://{{ hostname }}{{ item }}"
+    validate_certs: "{{ validate_certs }}"
+    ca_path: "{{ ca_path | default(omit) }}"
+    method: "{{ idrac_gather_facts_uri_method }}"
+    user: "{{ username }}"
+    password: "{{ password }}"
+    headers: "{{ idrac_gather_facts_uri_headers }}"
+    body_format: "{{ idrac_gather_facts_uri_body_format }}"
+    status_code: "{{ idrac_gather_facts_uri_status_code }}"
+    return_content: "{{ idrac_gather_facts_uri_return_content }}"
+  loop: "{{ collected_storage_entity }}"
+  register: controllers_url
+
+- name: Pick Controllers key value in one list
   ansible.builtin.set_fact:
-    diff_keys: "{{ controller_data.keys() | list | symmetric_difference((api_response.keys() | list)) }}"
+    controllers_list: "{{ controllers_list | default([]) + [item.json.Controllers] }}"
+  loop: "{{ controllers_url.results }}"
 
-- name: Set a diff of dict
+- name: Select list of values from dictionary
```

Comment:
Please mention which dictionary in name, seems generic.

Reply:
Done